src/software/test_util/test_util.cpp

```diff
@@ -102,4 +102,45 @@ namespace Test
         return Robot(robot_id_counter++, pt, Vector(), Angle(), AngularVelocity(),
                      Timestamp());
     }
+
+    bool TestUtil::checkIfVisuallySimilar(const Polygon &poly1, const Polygon &poly2)
+    {
+        auto ppoints1 = poly1.getPoints();
+        auto ppoints2 = poly2.getPoints();
+        if (ppoints1.size() != ppoints2.size())
+        {
+            return false;
+        }
+
+        for (int i = 0; i < ppoints1.size(); i++)
+        {
+            if (!checkIfVisuallySimilar(ppoints1[i], ppoints2[i]))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    bool TestUtil::checkIfVisuallySimilar(const Circle &circle1, const Circle &circle2)
+    {
+        if (!checkIfVisuallySimilar(circle1.getOrigin(), circle2.getOrigin()))
+        {
+            return false;
+        }
+        if (fabs(circle1.getRadius() - circle2.getRadius()) > METERS_PER_MILLIMETER)
+        {
+            return false;
+        }
+        return true;
+    }
+
+    bool TestUtil::checkIfVisuallySimilar(const Point &point1, const Point &point2)
+    {
+        return (point1.distanceFromPoint(point2) <=
+                (sqrt(2) * METERS_PER_MILLIMETER + GeomConstants::FIXED_EPSILON));
+    }
+
+    bool checkIfVisuallySimilar(const Circle &circle1, const Circle &circle2) {}
```

Comment:
So I think the concept of being able to check geometry with some specified tolerance is definitely a great idea. Here's a few tweaks we could make to help clarity a bit:
- `visuallySimilar` is very ambiguous, perhaps `checkGeometryEqualWithTolerance`?
- we should expose the tolerance as an argument in the functions, even if we assign it a default value, because there's lots of cases where we might want to change that, and I wouldn't want to presume a tolerance for every use case here.

Reply:
ok changed it to that